src/components/Card.js

```diff
@@ -1,51 +1,89 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
-    this._data = data;
+  constructor(
+    { data, handleImageClick, handleDeleteClick, handleLikeClick },
+    cardSelector
+  ) {
+    // Obtain the card's elements
+    this._name = data.name;
+    this._link = data.link;
+    this._cardId = data._id;
+
+    // Find the card template
     this._cardSelector = cardSelector;
+
+    // Functions
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
   }
 
   _setEventListeners() {
-    this._cardElement
-      .querySelector(".cards__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeIcon();
-      });
-    this._cardElement
-      .querySelector(".cards__delete-button")
-      .addEventListener("click", () => {
-        this._handleDeleteCard();
-      });
-    this._cardImageElement.addEventListener("click", () => {
-      this._handleImageClick(this._data);
-    });
+    // Add event listner for like button
+    this._likeButton.addEventListener("click", this._handleLikeClick);
+
+    // Add event listener for image
+    this._imageWindow = this._cardElement.querySelector(".cards__image");
+    this._imageWindow.addEventListener("click", () =>
+      this._handleImageClick({ link: this._link, name: this._name })
+    );
+
+    this._deleteButton.addEventListener("click", this._handleDeleteClick);
   }
 
-  _handleDeleteCard() {
-    this._cardElement.remove();
-    this._cardElement = null;
+  //Check if the card is liked
+  isLiked() {
+    return this._likeButton.classList.contains("cards__like-button_active");
+  }
+
+  //Set the card to liked or not
+  setLikes() {
+    this._renderLikes();
   }
 
-  _handleLikeIcon() {
+  // Change the like icon based on its status
+  _renderLikes() {
     this._cardElement
       .querySelector(".cards__like-button")
```

Comment:
You've already found `this._likeButton` as the like button in line 74. Please, use it everywhere

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.